### ISSUE TITLE
Indent guide in editor

### ISSUE THREAD

**Issue by lakshits11** [cpeditor/cpeditor#413]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
there should be indent guide lines so that we can see indentation properly like there in vs code.

**Describe alternatives you've considered**
none

**Additional context**

**Comment by coder3101** [comment]
It will be better, if you could attach a screenshot of the line you get in VSCode. 

I guess you mean, a vertical line from starting `{` to its closing `}` for indentations.

If yes, its hard to implement. In QScintilla, this is called `IndentationGuides`. SEE `#3` [here](https://qscintilla.com/#simple_options/indentation)

**Comment by lakshits11** [comment]
Yes, I mean that!!

**Comment by sonulohani** [comment]
Hi @coder3101 

Can i work on this feature if nobody is working.

Thanks

**Comment by sonulohani** [comment]
Hi @coder3101 Will https://github.com/cpeditor/cpeditor/pull/516 resolve this issue

**Comment by coder3101** [comment]
> Hi @coder3101 Will #516 resolve this issue

No. We will just replace QCodeEditor with KSyntaxHighlighting based Code Editor. It will have just code folding and the ability to choose custom themes.

**Comment by coder3101** [comment]
If you want, you can optimize the PR and your logic. Please note that the new code editor will be faster. The slow issue was attributed to some time taking code inside Editor line area, that has been resolved in both QCodeEditor and that PR.

**Comment by sonulohani** [comment]
> If you want, you can optimize the PR and your logic. Please note that the new code editor will be faster. The slow issue was attributed to some time taking code inside Editor line area, that has been resolved in both QCodeEditor and that PR.

The above points has been taken into consideration. Please review the above PR.
